docs/style/CODING_STYLE_GUIDE.md

```diff
@@ -0,0 +1,172 @@
+# Coding Style Guide
```

Comment:
Shall we present some of those conventions

- Variable, class, file naming?
- PascalCasing
- constexpr preferred to defines
- class enum
- Maybe some guidance for .gn file structure

Reply:
Tracking new content suggestions in an issue: https://github.com/project-chip/connectedhomeip/issues/36544